llvm/lib/Target/PowerPC/PPCTargetMachine.cpp

```diff
@@ -572,7 +572,8 @@ void PPCPassConfig::addMachineSSAOptimization() {
 void PPCPassConfig::addPreRegAlloc() {
   if (getOptLevel() != CodeGenOptLevel::None) {
     initializePPCVSXFMAMutatePass(*PassRegistry::getPassRegistry());
-    insertPass(VSXFMAMutateEarly ? &RegisterCoalescerID : &MachineSchedulerID,
+    insertPass(VSXFMAMutateEarly ? &TwoAddressInstructionPassID
```

Comment:
Just so I understand, we say two-address instruction pass here because the order of the passes originally were:
```
Two-Address instruction pass
Register Coalescing
PPC VSX FMA Mutation
```
But now you're. changing it so we're doing:
```
Two-Address instruction pass
PPC VSX FMA Mutation
Register Coalescing
```
right?

Or did I misunderstand?

Reply:
yes, you are correct .  
after the patch, when option -schedule-ppc-vsx-fma-mutation-early is set, 

```
Two-Address instruction pass
PPC VSX FMA Mutation
Register Coalescing
```

if the -schedule-ppc-vsx-fma-mutation-early is unset, it still  
```
Two-Address instruction pass
Register Coalescing
PPC VSX FMA Mutation
```